embedded-graphics/src/primitives/triangle/mod.rs

```diff
@@ -178,13 +148,119 @@ impl Triangle {
 
     /// Return the area of the triangle, doubled.
     ///
+    /// If the triangle's winding order is counter-clockwise, this method will return a negative
+    /// value.
+    ///
     /// This method can be used to determine if the triangle is colinear by checking if the returned
     /// value is equal to zero.
-    fn area_doubled(&self) -> i32 {
+    pub fn area_doubled(&self) -> i32 {
         let Self { p1, p2, p3 } = self;
 
         -p2.y * p3.x + p1.y * (p3.x - p2.x) + p1.x * (p2.y - p3.y) + p2.x * p3.y
     }
+
+    /// Create a new triangle with points sorted in a clockwise direction
+    pub fn sorted_clockwise(&self) -> Self {
+        match self.area_doubled().cmp(&0) {
+            // Triangle is wound CCW. Swap two points to make it CW.
+            Ordering::Less => Self::new(self.p2, self.p1, self.p3),
+            // Triangle is already CW, do nothing.
+            Ordering::Greater => *self,
+            // Triangle is colinear. Sort points so they lie sequentially along the line.
+            Ordering::Equal => {
+                let (p1, p2, p3) = sort_yx(self.p1, self.p2, self.p3);
+
+                Self::new(p1, p2, p3)
+            }
+        }
+    }
+
+    /// Find the center of gravity/centroid of the triangle
+    pub fn centroid(&self) -> Point {
+        (self.p1 + self.p2 + self.p3) / 3
+    }
+
+    /// Point inside triangle, ignoring pixels that partially lie outside triangle lines.
+    pub(self) fn mathematical_contains(&self, point: &Point) -> bool {
+        // Skip expensive calculations below if point is outside the bounding box
+        if !self.bounding_box().contains(*point) {
```

Comment:
You are skipping the bulk of this function, but not skipping the line walking in Triangle::contains(). Maybe it would be worth to move the bounding box check out so contains don't walk the edges when it doesn't have to?